test/GroupByTest.cpp

```diff
@@ -674,4 +676,69 @@ TEST_F(GroupBySpecialCount, computeGroupByForFullIndexScan) {
   // TODO<joka921> Add a test with only one column
 }
 
+namespace {
+// A helper function to set up expression trees in the following test.
+template <typename ExprT>
+auto make = [](auto&&... args) -> SparqlExpression::Ptr {
+  return std::make_unique<ExprT>(AD_FWD(args)...);
+};
+}  // namespace
+TEST(GroupBy, GroupedVariableInExpressions) {
+  parsedQuery::SparqlValues input;
+  using TC = TripleComponent;
+  /*
+   Test the following expression.
+   Note: The values are chosen s.t. the results are all integers. Otherwise we
+   would get into trouble with floating point comparisons. A check with a
+   similar query but with non-integral inputs and results can be found in the
+   E2E tests.
+
+   SELECT (AVG(?a + ?b) as ?x) (?a + COUNT(?b) AS ?y) WHERE {
+       VALUES (?x ?y) { (1.0 3.0) (1.0 7.0) (5.0 4.0)}
+   } GROUP BY ?x
+   */
```

Comment:
```suggestion
// Test the following SPARQL query:
//
// SELECT (AVG(?a + ?b) as ?x) (?a + COUNT(?b) AS ?y) WHERE {
//   VALUES (?x ?y) { (1.0 3.0) (1.0 7.0) (5.0 4.0)}
// } GROUP BY ?x
//
// Note: The values are chosen such that the results are all integers. Otherwise
// we would get into trouble with floating point comparisons. A check with a
// similar query but with non-integral inputs and results can be found in the
// E2E tests.
```